packages/app/src/overview/components/overview-screens/helpers/use-update-flow-state.hook.tsx

```diff
@@ -3,84 +3,125 @@
+      !needsForceUpdate &&
       osVersion
     ) {
-      const actionResult = checkForUpdate()
+      const actionResult = silentCheck()
       setSilentCheckForUpdatePromise(actionResult)
     }
 
     return () => {
       if (
         silentCheckForUpdatePromise &&
-        silentCheckForUpdateState === SilentCheckForUpdateState.Loading
+        silentCheckForUpdate === SilentCheckForUpdateState.Loading
       ) {
         silentCheckForUpdatePromise.abort()
         setSilentCheckForUpdatePromise(undefined)
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [silentCheckForUpdateState, forceUpdateNeeded, osVersion])
+  }, [silentCheckForUpdate, needsForceUpdate, osVersion])
 
   useEffect(() => {
-    if (forceUpdateNeeded) {
+    if (needsForceUpdate) {
       return
     }
     if (
-      (silentCheckForUpdateState === SilentCheckForUpdateState.Skipped &&
-        checkingForUpdateState === CheckForUpdateState.Initial) ||
-      silentCheckForUpdateState === SilentCheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.PerformedWithFailure
+      silentCheckForUpdate === SilentCheckForUpdateState.Loading ||
+      checkForUpdateState === CheckForUpdateState.Loading ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        silentCheckForUpdate === SilentCheckForUpdateState.Initial)
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+    } else if (allDownloaded) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Install)
+    } else if ((availableReleasesForUpdate || []).length > 0) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Download)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loading
+      error ||
+      [
+        CheckForUpdateState.PerformedWithFailure,
+        CheckForUpdateState.Failed,
+      ].includes(checkForUpdateState) ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        [
+          SilentCheckForUpdateState.Skipped,
+          SilentCheckForUpdateState.Failed,
+        ].includes(silentCheckForUpdate))
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.SilentCheckLoading)
-    } else if (checkingForUpdateState === CheckForUpdateState.Loading) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Performed
+      [CheckForUpdateState.Loaded, CheckForUpdateState.Performed].includes(
+        checkForUpdateState
+      ) ||
+      SilentCheckForUpdateState.Loaded === silentCheckForUpdate
     ) {
```

Comment:
Under this else-if the is no 'else', I am wondering that it might be a case when none of above condition is true, maybe we should set something when none of this cases is true? Or maybe not and leave it undefined...

Reply:
I think that a lot of issues with those badges were caused by the case where none of the conditions were true and then we defaulted to some text that did not match the situation. I think returning loading status or null would be nice (so that nothing will render based on those)